src/rdiff_backup/fs_abilities.py

```diff
@@ -415,6 +415,7 @@ def _detect_eas(self, rp, write):
             if write:
                 xattr.set(rp.path, b"user.test", test_ea)
                 read_ea = xattr.get(rp.path, b"user.test")
+                xattr.remove(rp.path, b"user.test")
```

Comment:
I did not read all the mail chain. So I understand to found the reason why the backup to NFS failed.

Getting xattr from NFS worked, but removing it from NFS did not work !? This is strange.

Reply:
Ha, you got me :smirk: this line has not much to do with the actual issue, it was just me thinking that the FS test would be more complete with trying as well to delete.
The actual issue was that the NFS sub-system would create an EA on a newly created file, which rdiff-backup would try to remove, which NFS wouldn't allow. The code was already foreseeing a few error codes, but NFS was sending EINVAL, which wasn't caught (even though IMHO an EACCES would have been more correct, but whatever). Now it's caught, all good.